operator/internal/manifests/object_storage.go

```diff
@@ -6,25 +6,37 @@ import (
 	appsv1 "k8s.io/api/apps/v1"
 )
 
-func configureDeploymentForStorageType(
+func configureDeploymentForStorageSpec(
```

Comment:
Does it make sense to move the two functions in `object_storage.go` to the `github.com/grafana/loki/operator/internal/manifests/storage`, replacing the current `ConfigureDeployment` and `ConfigureStatefulSet` functions? (the current `storage.Configure*` functions could be private then)